src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Extensions/SolutionExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Implementation.LanguageServer.Extensions
+{
+    internal static class SolutionExtensions
+    {
+        public static IEnumerable<Project> GetXamlProjects(this Solution solution)
+            => solution.Projects.Where(p => !p.SupportsCompilation && p.Language == StringConstants.XamlLanguageName);
```

Comment:
is the `!p.SupportsCompilation` check necessary? feels like checking language name is sufficient.

Reply:
good point. I guess I can remove the SupportsCompilation check.

---
In reply to: [516287248](https://github.com/dotnet/roslyn/pull/49145#discussion_r516287248) [](ancestors = 516287248)